src/main/java/edu/usfca/numberguessgame/service/GameService.java

```diff
@@ -60,84 +66,62 @@ public int verifyGuess(int target, int guess) {
      */
-    public String handleSetBound(String lowerBound, String upperBound, Model model) {
-
-        String boundResponse;
+    public Session handleSetBound(String lowerBound, String upperBound) {
 
         int lower = validateUserInput(lowerBound);
         int upper = validateUserInput(upperBound);
-        if(lower!=-1 && upper!=-1){
-
-            if (boundCheck(lower, upper)) {
-
-                int target = generateRandomInt(lower, upper);
 
-                Session session = new Session(target, lower, upper);
+        if(lower == -1 || upper == -1) {
+            // invalid bounds exception
+            throw new RuntimeException("Error! Make Sure You Entered Valid Bounds");
+        }
 
-                Session currId =  mongoTemplate.save(session);
+        if (!boundCheck(lower, upper)) {
+            // out of bounds exception
+            throw new RuntimeException("Error! Make Sure Upper Bound Is Greater Than Lower Bound");
+        }
 
-                model.addAttribute("currId",currId.get_id());
+        int target = generateRandomInt(lower, upper);
 
-                boundResponse= "Your Input Is Valid. Please Try To Guess It!";
+        Session session = new Session(target, lower, upper);
 
-            } else {
-                boundResponse= "Error! Make Sure Upper Bound Is Greater Than Lower Bound";
-            }
-        }else {
-            boundResponse= "Error! Make Sure You Entered Valid Bounds";
+        Session currId =  sessionRepository.save(session);
+        if (currId == null) {
+            throw new RuntimeException("Error saving new session data");
         }
 
-        if(boundResponse.equals("Your Input Is Valid. Please Try To Guess It!")) {
-            model.addAttribute("bound",boundResponse);
-            return "guess";
-        }else {
-            model.addAttribute("bound",boundResponse);
-            return "main";
-        }
+        return currId;
     }
 
     /**
      * message generator for Guess
      */
 
-    public String handleGuess(GuessRequest guessRequest, Model model) {
+    public Session handleGuess(String currId, String number) {
         int target;
         int parsedNumber;
         int compareResult;
 
-        String guessResponse;
-        model.addAttribute("currId",guessRequest.getCurrId());
-
-        String currId = guessRequest.getCurrId();
+        Optional<Session> session = sessionRepository.findById(currId);
 
-        Session session = mongoTemplate.findById(currId, Session.class);
-
-        if(session == null) {
-            return "Error! The userId is not found";
+        if(session.isEmpty()) {
+            throw new RuntimeException("Error! The userId is not found");
         }
-        target = session.getTarget();
+        target = session.get().getTarget();
 
-        parsedNumber = validateUserInput(guessRequest.getNumber());
+        parsedNumber = validateUserInput(number);
         compareResult = verifyGuess(target, parsedNumber);
```

Comment:
Again this method is redundant. You're using the same logic twice in a row. Just copy what's in `verifyGuess` method below keeping the same throws. Other than that method looks good!